File: projecteuler/PE_119/Python/PE_119.py

def digit_sum(number):
    return sum(map(int, str(number)))

def PE_119(n):
    index = 0
    total = 0
    digit_power_sums = []
    for k in range(2, 101):
        for j in range(2, 101):
            number = k**j
            if digit_sum(number) == k:
                digit_power_sums.append(number)
    return sorted(digit_power_sums)[n-1]
print(PE_119(30))
# isDigitPowerSum([1, 0, 0, 1, 2, 3, 3, 3, 3, 3, 4, 4, 4, 5, 9, 9])

File: projecteuler/PE_119/Julia/PE_119.jl
function digitSum(number)
    return sum(digits(number))
end

function isPowerDigitSum(number)
    power = log(digitSum(number), number)
    return ceil(power) == floor(power)
end

println(isPowerDigitSum(512))
println(6^2, ", ", 6^3)

total = 0
for i = 1:10^10
    if isPowerDigitSum(i)
        total += 1
        println(i, ", ", total)
    end
end
